submissions/SerhiiYakovenko/js-dom/script.js

```diff
@@ -0,0 +1,128 @@
+const items = [
+  {
+    heading: "City at day",
+    img: "img/city.jpg",
+    text:
+      "Enter the massive open world of Night City, a place that sets new standards in terms of visuals, complexity and depth.",
+  },
+  {
+    heading: "City at night",
+    img: "img/night-city.jpg",
+    text: `Cyberpunk 2077 is an open-world, action-adventure story set in Night City, a megalopolis obsessed with power, glamour and body modification.
+        You play as V, a mercenary outlaw going after a one-of-a-kind implant that is the key to immortality.
+        You can customize your character’s cyberware, skillset and playstyle, and explore a vast city where the choices you make shape the story and the world around you.`,
+  },
+  {
+    heading: "Mercenary",
+    img: "img/mercenary.jpg",
+    text:
+      "Become a cyberpunk, an urban mercenary equipped with cybernetic enhancements and build your legend on the streets of Night City.",
+  },
+  {
+    heading: "Implants",
+    img: "img/implant.jpg",
+    text:
+      "Take the riskiest job of your life and go after a prototype implant that is the key to immortality.",
+  },
+];
+
+const body = document.querySelector("body");
+
+// Header
+const header = document.createElement("header");
+let h1 = document.createElement("h1");
+h1.textContent =
+  "Welcome to the CyberDOM. Here you can get more info about Cyberpunk 2077 game.";
+header.appendChild(h1);
+body.appendChild(header);
+
+// Nav and Main
+const nav = document.createElement("nav");
```

Comment:
you can just use template strings and build all of your html elements there. 
Then you can use only ONE insert to the DOM in order to append this template string with some content to some element. It will help you to increase the performance of your app and make your code cleaner

Reply:
So I was trying to implement something like this :
```
nav.innerHtml = `<ul class="nav__items">`;

articles.forEach((article) => {
  nav.innerHtml += `<li class="nav__item"> ${article.heading} </li>`;
  main.insertAdjacentHTML("afterbegin",
  `<div class="main__item ${article.heading.toLowerCase().split(" ").join("_")} hidden">
    <p class="main__content">${article.text}</p>
    <img class="main_image" src="${article.img}" alt="${article.img.replace("img/", "").replace(".jpg", "")}"></img>
  </div>`)
});
nav.innerHtml += `</ul>`;
```
To have nav section in ht end like this 
```
<nav class="nav"><ul class="nav__items"><li class="nav__item"> Implants </li><li class="nav__item"> Mercenary </li><li class="nav__item"> City at night </li><li class="nav__item"> City at day </li></ul></nav>
```
But with no success, so I left implementation like this:
create `<ul>` element and generate `<li>` items by code.

Also in the end I used this `NavList` to add event listener on it to prevent this error:
`script.js:82 Uncaught TypeError: Cannot read property 'classList' of undefined
    at HTMLElement.<anonymous>`
So I would be happy to hear how I can rewrite it to make it work as you see it.

Reply:
In your suggested approach you used too many DOM operations as well.
As I see you used the `innerHtml` and the `insertAdjacentHTML` on each loop iteration. It can decrease the performance of your app.
You can consider this approach, for example:
``` javascript
let someContent = '';
articles.forEach(({heading, img, text}) => {
  someContent += `<li>
     <h3>${heading}</h3>
     <img alt="${heading}" src={img}/>
     <p>${text}</p>
    </li>`
});
const navigationWrap = document.querySelector('.someUlClassName');
navigationWrap.innerHTML = someContent;
// or navigationWrap.insertAdjacentHTML("afterbegin", someContent);
```

About the error which you got. 
Don't forget about using `debugger;` or just simple `console.log()` in your code.